File: SDKExamples/Chart/src/main/java/fragments/series/ScatterPointSeriesFragment.java
package fragments.series;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.telerik.manual.tests.R;
import com.telerik.widget.chart.engine.databinding.FieldNameDataPointBinding;
import com.telerik.widget.chart.visualization.cartesianChart.RadCartesianChartView;
import com.telerik.widget.chart.visualization.cartesianChart.axes.CategoricalAxis;
import com.telerik.widget.chart.visualization.cartesianChart.axes.LinearAxis;
import com.telerik.widget.chart.visualization.cartesianChart.series.categorical.SplineSeries;
import com.telerik.widget.chart.visualization.cartesianChart.series.scatter.ScatterPointSeries;

import java.util.ArrayList;
import java.util.Random;

/**
 * Created by ginev on 11/18/2014.
 */
public class ScatterPointSeriesFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        FrameLayout rootView = (FrameLayout)inflater.inflate(R.layout.fragment_scatter_series, container, false);

        rootView.addView(this.createChart());
        return rootView;
    }

    private RadCartesianChartView createChart(){
        //Create the Chart View
        RadCartesianChartView chart = new RadCartesianChartView(this.getActivity());

        //Create the bar series and attach axes and value bindings.
        ScatterPointSeries scatterPointSeries = new ScatterPointSeries();


        scatterPointSeries.setXValueBinding(new FieldNameDataPointBinding("value1"));
        scatterPointSeries.setYValueBinding(new FieldNameDataPointBinding("value2"));

        LinearAxis verticalAxis = new LinearAxis();
        //The values in the linear axis will not have values after the decimal point.
        verticalAxis.setLabelFormat("%.2f");
        LinearAxis horizontalAxis = new LinearAxis();
        horizontalAxis.setLabelFormat("%.2f");
        scatterPointSeries.setVerticalAxis(verticalAxis);
        scatterPointSeries.setHorizontalAxis(horizontalAxis);

        //Bind series to data
        scatterPointSeries.setData(this.getData());

        //Add series to chart
        chart.getSeries().add(scatterPointSeries);
        return chart;
    }

    private ArrayList<ScatterDataEntity> getData(){
        Random numberGenerator = new Random();
        ArrayList<ScatterDataEntity> result = new ArrayList<ScatterDataEntity>(8);

        for (int i = 0; i < 20; i++){
            ScatterDataEntity entity = new ScatterDataEntity();
            entity.value1 = numberGenerator.nextDouble() * 10;
            entity.value2 = numberGenerator.nextDouble() * 10;
            result.add(entity);
        }

        return result;
    }

    public class ScatterDataEntity{
        public double value1;
        public double value2;
    }

}

File: Samples-Xamarin/Fragments/Chart/Series/ScatterPointSeriesFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Java.Util;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart.Series.Scatter;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart.Axes;
using Com.Telerik.Widget.Chart.Engine.Databinding;

namespace Samples
{
	public class ScatterPointSeriesFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			ViewGroup rootView = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_chart_example, container, false);
			rootView.AddView(this.createChart());
			return rootView;
		}


		private RadCartesianChartView createChart(){
			//Create the Chart View
			RadCartesianChartView chart = new RadCartesianChartView(this.Activity);

			//Create the scatter point series and attach axes and value bindings.
			ScatterPointSeries scatterPointSeries = new ScatterPointSeries();


			scatterPointSeries.XValueBinding = new XValueBinding();
			scatterPointSeries.YValueBinding = new YValueBinding();

			LinearAxis verticalAxis = new LinearAxis();
			//The values in the linear axis will not have values after the decimal point.
			verticalAxis.LabelFormat = "%.2f";
			LinearAxis horizontalAxis = new LinearAxis();
			horizontalAxis.LabelFormat = "%.2f";
			scatterPointSeries.VerticalAxis = verticalAxis;
			scatterPointSeries.HorizontalAxis = horizontalAxis;

			//Bind series to data
			scatterPointSeries.Data = this.getData();

			//Add series to chart
			chart.Series.Add(scatterPointSeries);
			return chart;
		}

		private ArrayList getData(){
			Java.Util.Random numberGenerator = new Java.Util.Random();
			ArrayList result = new ArrayList(8);

			for (int i = 0; i < 20; i++){
				ScatterDataEntity entity = new ScatterDataEntity();
				entity.value1 = numberGenerator.NextDouble() * 10;
				entity.value2 = numberGenerator.NextDouble() * 10;
				result.Add(entity);
			}

			return result;
		}



		public String Title() {
			return "Scatter point series";
		}

		public class XValueBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				ScatterDataEntity entity = (ScatterDataEntity)p0;
				return entity.value1;
			}

		}

		public class YValueBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				ScatterDataEntity entity = (ScatterDataEntity)p0;
				return entity.value2;
			}

		}

		public class ScatterDataEntity : Java.Lang.Object{
			public double value1;
			public double value2;
		}
	}
}

